logstash-core/lib/logstash/util/cloud_setting_id.rb

```diff
@@ -0,0 +1,37 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "base64"
+
+module LogStash module Util class CloudSettingId
+  attr_reader :original, :decoded, :label, :elasticsearch_host, :kibana_host
+
+  def initialize(value)
+    return if value.nil?
+
+    unless value.is_a?(String)
+      raise ArgumentError.new("Cloud Id must be String. Received: #{value.class}")
+    end
+    @original = value
+    @label, sep, last = value.partition(":")
+    if last.empty?
+      @decoded = Base64.urlsafe_decode64(@label) rescue ""
```

Comment:
We should not ignore the exception, here. If a user makes a mistake, it'd be nice to surface the problem (invalid base64, in this case)